engine/runtime-compiler/src/main/java/org/enso/compiler/MetadataInteropHelpers.java

```diff
@@ -0,0 +1,42 @@
+package org.enso.compiler;
+
+import java.util.Optional;
+import org.enso.compiler.core.IR;
+import org.enso.compiler.core.ir.ProcessingPass;
+import org.enso.compiler.pass.IRPass;
+import scala.Option;
+
+/**
+ * A set of helper methods for handling of IR metadata.
+ *
+ * <p>This encapsulates the friction of interop between Scala and Java types.
+ */
+public class MetadataInteropHelpers {
+  public static <T> Optional<T> getOptionalMetadata(IR ir, IRPass pass, Class<T> expectedType) {
```

Comment:
This wont' work, because `getMetadata` does not return `null`, it actually _throws_ an exception.

This is deliberate - if I expect the metadata _must_ be present, I use `getMetadata` which will fail with an exception if my assertion is false, and if I am requesting metadata that _may or may not be available_ I use `getOptionalMetadata` to indicate that.

If `getMetadata` did not throw, we would get an NPE down the line that is harder to debug.

Reply:
`Optional.of(getMetadataOrNull(...))` be it then!